clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBackedSchemaReader.java

```diff
@@ -187,12 +191,20 @@ public Schema getKeySchema() {
   public Schema getValueSchema(int id) {
     // Should call with refresh as any router transient error can lead to null schema stored in the map
     // `valueSchemaEntryMap`
-    SchemaEntry valueSchemaEntry = maybeFetchValueSchemaEntryById(id, true);
-    if (!isValidSchemaEntry(valueSchemaEntry)) {
-      LOGGER.warn("Got null value schema from Venice for store: {} and id: {}", storeName, id);
+    Schema valueSchema;
+    try {
+      valueSchema = RetryUtils.executeWithMaxAttemptAndExponentialBackoff(() -> {
+        SchemaEntry schemaEntry = maybeFetchValueSchemaEntryById(id, true);
+        if (!isValidSchemaEntry(schemaEntry)) {
+          throw new VeniceException("Got null value schema from Venice for store " + storeName + " and id " + id);
+        }
+        return schemaEntry.getSchema();
+      }, 5, Duration.ofMillis(10), Duration.ofMillis(500), Duration.ofSeconds(5), RETRY_FAILURE_TYPES);
```

Comment:
would it be only 10, 20, 40, 80, 160 millisecond delay in total? Do we know what's the transient failure that fail the schema retrieval? I wonder if this is enough for retry.